src/hypothesis/internal/compat.py

```diff
@@ -502,10 +502,16 @@ def implements_iterator(it):
 
 if PY2:
     def floor(x):
-        return int(math.floor(x))
+        # `int(math.floor(x))` fails for negative long integers
```

Comment:
(I see this is copied from the old code in `integers()`, but I’m wondering if you’ve seen anything more detailed.)